### ISSUE TITLE
Library does not fetch all the repositories for a team

### ISSUE THREAD

**Issue by AlphaGit** [gastonmancini/bitbucket-opened-pullrequests-report#1]
The library does not seem to be fetching all repositories for a team. Following the steps that the library executes, I was able to find that the results from `bitbucketClient.repo.fetch(teamName)` are paginated.

Here's an example:

```javascript
bitbucketclient.repo.fetch("[TEAM NAME REDACTED]").then(console.log);
```

```javascript
{ pagelen: 10,
  size: 22,
  values:
   [ { scm: 'git',
       website: '',
       has_wiki: false,
       name: '[PROJECT NAME REDACTED]',
       links: [Object],
       fork_policy: 'no_public_forks',
       uuid: '{a353dcd0-4f8a-43cd-9db3-7ec76fcbd82e}',
       project: [Object],
       language: 'c#',
       created_on: '2015-02-09T02:01:40.713252+00:00',
       full_name: '[REDACTED]',
       has_issues: false,
       owner: [Object],
       updated_on: '2016-12-15T14:28:15.386384+00:00',
       size: 356330481,
       type: 'repository',
       slug: '[PROJECT NAME REDACTED]',
       is_private: true,
       description: '[REDACTED]' },
... ],
  page: 1,
  next: 'https://api.bitbucket.org/2.0/repositories/[TEAM NAME REDACTED]?page=2' }
```

Note how the response provides:

- The current page number (`page`, 1 in this example)
- The amount of elements per page (`pagelen`, 10 in this example)
- The total size of the results pool (`size`, 22 in this example)

These are some approaches I thought of:

- Continue calling until there are no pages left
- Set a page size long enough (maybe configurable?) so that everything fits in one page

**Comment by AlphaGit** [comment]
Note: it seems that [bitbucketjs](https://bitbucket.org/atlassian/bitbucketjs) does not support pagination options for fetching repositories. ([Source](https://bitbucket.org/atlassian/bitbucketjs/src/abffce6e69aa85224e4aad9db75c38284c8b9220/src/repo.js?at=master&fileviewer=file-view-default))

If I have the chance, I will fork them and submit a PR to them with those options enabled.

**Comment by AlphaGit** [comment]
After further investigation, turns out that [BitBucket 2.0 API does not support indicating pagination directly](https://developer.atlassian.com/bitbucket/api/2/reference/meta/pagination). They require the clients to go through the pagination links.

I had modified the bitbucketjs package (changes can be seen [here](https://bitbucket.org/AlphaBit2001/bitbucketjs/commits/5bac87ea735f24d0ab43b252340d3ccfadad0c6c)), and upon testing it I found this. With this new information, I believe my changes don't make sense anymore, so I won't be sending a PR to the Attlassian team.

I'll proceed with our first approach of iterating through pages.
